File: python-impl/insfrastructure/usecases/get_account_by_id_usecase.py
from insfrastructure.repositories.bank_account_repository import BankAccountRepository


class GetAccountByIdUseCase:
    def __init__(self, bank_account_repo: BankAccountRepository):
        self.bank_account_repo = bank_account_repo

    def execute(self, account_id: int) -> bool:
        return self.bank_account_repo.find_by_id(account_id)
        

File: typescript-impl/src/infrastructure/usecases/get_account_by_id_usecase.ts
import { BankAccountRepository } from '../repositories/bank_account_repository';
import { BankAccountEntity } from '../entities/bank_account_entity';

export class GetAccountByIdUseCase {
    constructor(private bank_account_repo: BankAccountRepository) { }

    execute(account_id: number): BankAccountEntity | null {
        return this.bank_account_repo.find_by_id(account_id);
    }
}
